### ISSUE TITLE
Device no longer working

### ISSUE THREAD

**Issue by tkroon89** [bisand/net.biseth.myenergi#56]
Hi!

Since two or three days, the device in the app stops working. When I soft reset the myenergi hub, it starts working for a short while, but then fails again shorty. Also restarting the myenergi app on Homey does not resolve anything.

This has never happened before, and I haven't changed anything. The Zappi v5 firmware update was a couple of weeks ago, not a couple of days. Also Homey was not updated.

The myenergi app on my iPhone does work properly, indicating the Zappi + hub work as expected. Do you have any idea? Thank you in advance, and let me know if I can assist you with anything.

Diagnostics: e88a56b8-2cc1-4cd8-a3c6-9a66c99ac42f

![IMG_6079](https://github.com/bisand/net.biseth.myenergi/assets/60198451/b913205c-fd33-466b-8e7e-db5eaebfa4ab)
![IMG_6078](https://github.com/bisand/net.biseth.myenergi/assets/60198451/3433b303-8c63-4aa4-a782-c4b6d74b28ac)

**Comment by SmartHomeEnthusiast** [comment]
Have you tried removing the device from Homey and then re-adding?

**Comment by tkroon89** [comment]
Yes, I have. On all three Homey Pro’s (two old Pro’s and the new 2023) I encounter the same issue. Deleting and adding the device again does not solve the issue. Could you find any clues in the diagnostics?

**Comment by SmartHomeEnthusiast** [comment]
I'm not the dev so have no access to the diagnostic, pretty sure I couldn't help at that level anyway.
Have you contacted myEnergi? might it be your api key that should be changed? Might be worth trying that anyway

**Comment by tkroon89** [comment]
Thanks for thinking with me. I’ll wait for the developer before contacting myenergi, since the myenergi app on my iPhone is working fine.

**Comment by bisand** [comment]
I think the problem is caused by an unhandled rejection somewhere in the app code. The diagnostics report contains some information about that. I will take a look at it and publish a new version as soon as I have a fix for it.
One of the exceptions states "Invalid Capability: meter_power", so you can first try to change/remove any flows using that capability/value.

I will post more info here when I know more.

**Comment by tkroon89** [comment]
Thank you. On two Homey's I don't have any flows. 
Additional note: restarting the Hub does not result in a working Homey device anymore. Also if I delete and add the Zappi again.

Please let me know when I can help you.

**Comment by tkroon89** [comment]
I have done some testing. Check te video.

While I tap the charge mode STOP, it seems like it's paused and waiting for some (?) surplus like it's in an ECO-mode. I'm not sure which charging state the Zappi is reporting, but I think you could see that in the diagnostics. 

When I manually turned Zappi ON to the FAST charging mode and restarted the Homey-app, it worked again. So the error you generated in the screenshot shared in my first post seems indeed correct. I will also report this to myenergi, because it seems strange I am the only one with this issue.

https://github.com/bisand/net.biseth.myenergi/assets/60198451/22bf1ad7-4430-4736-9e9d-32d435431e03

**Comment by bisand** [comment]
Closing this as part of a housekeeping sweep. This report is from app v1.9-era code, and the app has been substantially reworked since (v1.10, v1.11 with Energy Hub, and a batch of fixes landing in the next release).

Relevant to this exact symptom: we recently found and fixed a crash in the Zappi device's initialization that would leave the device permanently broken (no updates, no controls) until an app restart — which matches what you described very closely. The fix is in the upcoming release.

@tkroon89 since you're also the reporter of #37: the next test version contains both that fix and the new phase switching support, so you can verify both at once. If the device still misbehaves on the new version, please open a fresh issue with a new diagnostics report.

**Comment by bisand** [comment]
Test version v2.0.0 is available (https://homey.app/a/net.biseth.myenergi/test/) and contains the initialization crash fix mentioned above — if the device misbehaves again on this version, please open a fresh issue with a diagnostics report.
